Added command line option for specifying the access modifier for generated java classes

avf opened the pull request:
This pull request adds the feature requested in issue #245. Please be gentle, this is my first time writing Scala code. Let me know if I can make any improvements before merging.

smarx commented:
*Automated message from Dropbox CLA bot*

@avf, it looks like you've already signed the Dropbox CLA. Thanks!

artwyman commented on src/source/generator.scala:
```diff
@@ -137,6 +138,22 @@ package object generatorTools {
 
+  object JavaAccessModifier extends Enumeration {
+    val Public = Value("public")
+    val Protected = Value("protected")
```
I don't think that "protected" is a valid value for top-level classes.  It would be meaningless, so I think it's disallowed.

http://stackoverflow.com/questions/3869556/why-a-class-cannot-be-defined-as-protected

I think public and package-private are the only values this option needs to support.

artwyman commented on src/source/generator.scala:
```diff
@@ -137,6 +138,22 @@ package object generatorTools {
+      javaAccessModifier match {
+        case Public => "public "
+        case Protected => "protected "
+        case Package => ""
```
In other cases where package-private is used by Djinni (e.g. in constant fields) the generator outputs a comment /*package*/ in place of the accessibility.

artwyman commented:
I'm not really a guru on Scala style either, but this looks pretty good to me.  I made some suggestions in line comments.

I'd like to see the new argument exercised in the test-suite or the example, so we have some validation that it works, and can see the effect on the generated code.  Ideally, I'd suggest changing one of the two to use a non-default value (package), while the other uses the default value, so both can be seen in action by someone examining code in the repo.  A quick test suggests that the example will still compile unchanged with package-private, since it's all one package.  The test-suite doesn't, since tests are in a separate package.

avf commented:
Hi @artwyman, thanks for your feedback!
I've implemented your suggestions and removed the `protected` option, added `/*package*/` to the generator and updated the sample to use the `package` option.

artwyman commented:
LGTM.  Thanks @avf.
